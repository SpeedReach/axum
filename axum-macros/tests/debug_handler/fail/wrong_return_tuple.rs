
#[axum::debug_handler]
async fn handler() -> (
    axum::http::StatusCode,
    axum::Json<&'static str>,
    axum::response::AppendHeaders<[( axum::http::HeaderName,&'static str); 1]>,
) {
    panic!()
}

fn main(){}