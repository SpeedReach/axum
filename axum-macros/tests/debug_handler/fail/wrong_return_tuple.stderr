error: This type only implements IntoResponse but not IntoResponseParts, try moving it to the last element
 --> tests/debug_handler/fail/wrong_return_tuple.rs:6:5
  |
6 |     axum::Json<&'static str>,
  |     ^^^^^^^^^^^^^^^^^^^^^^^^

error[E0277]: the trait bound `CustomIntoResponse: IntoResponseParts` is not satisfied
  --> tests/debug_handler/fail/wrong_return_tuple.rs:24:5
   |
24 |     CustomIntoResponse,
   |     ^^^^^^^^^^^^^^^^^^ the trait `IntoResponseParts` is not implemented for `CustomIntoResponse`
   |
   = help: the following other types implement trait `IntoResponseParts`:
             (T1, T2)
             (T1, T2, T3)
             (T1, T2, T3, T4)
             (T1, T2, T3, T4, T5)
             (T1, T2, T3, T4, T5, T6)
             (T1, T2, T3, T4, T5, T6, T7)
             (T1, T2, T3, T4, T5, T6, T7, T8)
             (T1, T2, T3, T4, T5, T6, T7, T8, T9)
           and $N others
   = help: see issue #48214
   = help: add `#![feature(trivial_bounds)]` to the crate attributes to enable
